src/main/webapp/view/transcript/transcript.js

```diff
@@ -115,60 +86,108 @@ function seekTranscript(currentTime) {
   if (currentTimeMs < currentTranscriptLine.startTimestampMs) {
     return;
   }
-  if (isWithinCurrentTimeRange(currentTimeMs)) {
-    addBold(currentTranscriptLine);
+  if (currentTranscriptLine.isWithinCurrentTimeRange(currentTimeMs)) {
+    currentTranscriptLine.addBold();
     return;
   }
-  removeBold(currentTranscriptLine);
+  currentTranscriptLine.removeBold();
   currentTranscriptLine = currentTranscriptLine.nextElementSibling;
-  scrollToTopOfTranscript(currentTranscriptLine);
-  addBold(currentTranscriptLine);
+  currentTranscriptLine.scrollToTopOfTranscript();
+  currentTranscriptLine.addBold();
   // TODO: Handle the case where the video isn't only playing.
 }
 
 /**
- * Bolds the text in `transcriptLineLiElement` if it is not already
- * bolded.
+ * Creates a transcript line element containing the text,
+ * start time, and end time from `transcriptLine`.
  */
-function addBold(transcriptLineLiElement) {
-  if (isBolded(transcriptLineLiElement)) {
-    return;
+class TranscriptLine extends HTMLElement {
+  /**
+   * Creates a custom HTML element representing a transcript line.
+   *
+   * <p>Uses the template and slots defined in `TRANSCRIPT_TEMPLATE` to
+   * help create the transcript line.
+   *
+   * @param transcriptLine The transcriptLine from `ENDPOINT_TRANSCRIPT`
+   *     whose `attributes` should be used.
+   */
+  constructor(transcriptLine) {
+    super();
+    const timestampRange = window.createTimestampRange(
+        transcriptLine.startTimestampMs, transcriptLine.endTimestampMs);
+
+    const template = document.getElementById(TRANSCRIPT_TEMPLATE);
+    this.attachShadow({mode: 'open'});
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+    this.updateSlot('timestamp-range', timestampRange, STYLE_TIME_RANGE);
+    this.updateSlot('content', transcriptLine.content, STYLE_CONTENT);
+
+    this.startTimestampMs = transcriptLine.startTimestampMs;
+    this.endTimestampMs = transcriptLine.endTimestampMs;
+    // Sets the current transcript line to be the first line.
+    if (currentTranscriptLine == null) {
+      currentTranscriptLine = this;
+    }
   }
-  transcriptLineLiElement.classList.add(BOLD_FONT_WEIGHT);
-  transcriptLineLiElement.classList.remove(DEFAULT_FONT_WEIGHT);
-}
 
-/**
- * Removes bold from the text in `transcriptLineLiElement` if it
- * is currently bolded.
- */
-function removeBold(transcriptLineLiElement) {
-  if (!isBolded(transcriptLineLiElement)) {
-    return;
+  /**
+   * Updates the template slot `slotName` with `slotValue` and styling
+   * from `slotStyle`.
+   */
+  updateSlot(slotName, slotValue, slotStyle) {
```

Comment:
nit: maybe rename this to `updateTemplateSlot`

Reply:
done, thanks!